### PR TITLE
autolock tech debt cleanup

### LINKED ISSUE #303: Address tech debt from autolocking
Two main components:
- [ ] Address nits & comments from @linuxwolf and @jhugman review on #240
- [ ] Fixup the `AutoLockTest`s so that they properly integration test the autolock behavior

### PR DESCRIPTION
**Pull request by sashei**
Fixes #303 

## To Do

- [x] double check the original issue to confirm it is fully satisfied
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

File: app/src/main/java/mozilla/lockbox/store/AutoLockStore.kt [hunk 1]
```diff
@@ -41,7 +41,7 @@ class AutoLockStore(
 
     var lockingSupport: LockingSupport = SystemLockingSupport()
 
-    val lockRequired: Observable<Boolean> = ReplaySubject.create()
+    val lockRequired: Observable<Boolean> = ReplaySubject.createWithSize(1)
```
Comment (sashei) [inline app/src/main/java/mozilla/lockbox/store/AutoLockStore.kt]: probably 6 hours of debugging just to change this line, y'all >.<

File: app/src/androidTest/java/mozilla/lockbox/Navigator.kt [hunk 2]
```diff
@@ -28,12 +29,16 @@ import mozilla.lockbox.robots.securityDisclaimer
...
+import mozilla.lockbox.view.RootActivity
 import org.junit.Assert
 
 @ExperimentalCoroutinesApi
 class Navigator {
-    fun resetApp() {
-        Dispatcher.shared.dispatch(LifecycleAction.UseTestData)
+    fun resetApp(activityRule: ActivityTestRule<RootActivity>) {
+        activityRule.activity.runOnUiThread {
+            activityRule.activity.recreate()
```
Comment (jhugman) [inline app/src/androidTest/java/mozilla/lockbox/Navigator.kt]: 😻

File: app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt [hunk 3]
```diff
@@ -58,7 +58,7 @@ class RoutePresenter(
     override fun onViewReady() {
         navController = Navigation.findNavController(activity, R.id.fragment_nav_host)
 
-        val lockObservable = accountStore.syncCredentials
+        val autoLockActions = accountStore.syncCredentials
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/presenter/RoutePresenter.kt]: 💪

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 4]
```diff
@@ -178,7 +178,9 @@ open class DataStore(
             support.syncConfig = SyncUnlockInfo(kid, accessToken.token, syncKey, tokenServerURL)
         }
 
-        if (!credentials.isNew) return
+        if (!credentials.isNew) {
+            return
+        }
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: 😄

File: app/src/main/java/mozilla/lockbox/view/RootActivity.kt [hunk 5]
```diff
@@ -14,7 +14,7 @@ import mozilla.lockbox.presenter.RoutePresenter
 
 @ExperimentalCoroutinesApi
 class RootActivity : AppCompatActivity() {
-    internal var presenter: RoutePresenter = RoutePresenter(this)
+    private var presenter: RoutePresenter = RoutePresenter(this)
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/view/RootActivity.kt]: Is this still needed?

**Pull request merged**